### PR TITLE
Re-enables errorprone on JDK 10+

### PR DESCRIPTION
**Pull request by adriancole**
This uses the commandline instructions here:
  http://errorprone.info/docs/installation

See https://github.com/google/error-prone/issues/860

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -640,6 +639,46 @@
...
+                  <compilerArgs>
+                    <arg>-XDcompilePolicy=simple</arg>
+                    <arg>-Xplugin:ErrorProne</arg>
+                    <!-- TODO: why does this not work? -->
+                    <!--<arg>-XepDisableWarningsInGeneratedCode</arg>-->
+                  </compilerArgs>
+                  <annotationProcessorPaths>
+                    <processorPath>
+                      <groupId>com.google.errorprone</groupId>
+                      <artifactId>error_prone_ant</artifactId>
```
Comment (adriancole) [inline pom.xml]: do we really need the ant classifier? or will core work?

File: pom.xml [hunk 2]
```diff
@@ -640,6 +639,46 @@
...
+                <goals>
+                  <goal>compile</goal>
+                </goals>
+                <configuration>
+                  <forceJavacCompilerUse>true</forceJavacCompilerUse>
+                  <compilerArgs>
+                    <arg>-XDcompilePolicy=simple</arg>
+                    <arg>-Xplugin:ErrorProne</arg>
+                    <!-- TODO: why does this not work? -->
+                    <!--<arg>-XepDisableWarningsInGeneratedCode</arg>-->
```
Comment (adriancole) [inline pom.xml]: this crashes when uncommented:

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] javac: invalid flag: -XepDisableWarningsInGeneratedCode
Usage: javac <options> <source files>
use --help for a list of possible options
warning: [options] bootstrap class path not set in conjunction with -source 1.8
```

**Comment by adriancole** [comment]
Works with the problem of crashing on args (commented about this)

```bash
$ ./mvnw clean compile -pl brave
--snip--
[INFO] --- maven-compiler-plugin:3.8.0:compile (default-compile) @ brave ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 50 source files to /Users/acole/oss/brave/brave/target/classes
[WARNING] [options] bootstrap class path not set in conjunction with -source 1.8
[WARNING] WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.google.errorprone.bugpatterns.FutureReturnValueIgnored (file:/Users/acole/.m2/repository/com/google/errorprone/error_prone_ant/2.3.1/error_prone_ant-2.3.1.jar) to field com.sun.tools.javac.code.Type$StructuralTypeMapping$4.this$0
WARNING: Please consider reporting this to the maintainers of com.google.errorprone.bugpatterns.FutureReturnValueIgnored
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
/Users/acole/oss/brave/brave/src/main/java/brave/propagation/ThreadLocalSpan.java:[99,6] [ThreadLocalUsage] ThreadLocals should be stored in static fields
    (see http://errorprone.info/bugpattern/ThreadLocalUsage)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 4.289 s
[INFO] Finished at: 2018-08-20T09:38:30+08:00
[INFO] ------------------------------------------------------------------------
```

**Comment by adriancole** [comment]
oops also need to figure our how to disable this in certain sub-projects, like JMH..

File: pom.xml (hunk 2 above)
Comment (michaelhixson) [reply pom.xml]: You need to put all the Error Prone flags in the same `<arg>` as `-Xplugin:ErrorProne` like
```
<arg>-Xplugin:ErrorProne -XepDisableWarningsInGeneratedCode</arg>
```

Here's an example from another project I work on that uses Error Prone as a compiler plugin:  https://github.com/TechEmpower/tfb-status/blob/e3e1c5d2684252d2ad585545016c98574bf78e81/pom.xml#L451-L534

File: pom.xml (hunk 1 above)
Comment (michaelhixson) [reply pom.xml]: `error_prone_core` should work.

**Comment by adriancole** [comment]
circleci is angry for unrelated reasons

**Pull request merged**